pkg/controller/certificates/certificate_request.go

```diff
@@ -483,6 +483,44 @@ func (c *certificateRequestManager) processCertificate(ctx context.Context, crt
 		}
 	}
 
+	// If the CertificateRequest exists but has failed then we check the failure
+	// time. If the failure time doesn't exist or is over an hour in the past
+	// then delete the request so it can be re-created on the next sync. If the
+	// failure time is less than an hour in the past then schedule this owning
+	// Certificate for a re-sync in an hour.
+	if existingReq != nil && apiutil.CertificateRequestHasFailed(existingReq) {
```

Comment:
Instead of checking if `existingReq != nil` here, we can move this whole block to after the point where we check `if existingReq == nil` and then only check for the Reason.

Thinking about it, we don't actually have a point where we check the `status` is not Pending either - we could probably switch here on the Ready condition's status:

* `Ready`: decode cert (i.e. existing flow after this block)
* `Pending`: wait
* `Failed`: this logic

Reply:
:heart: